Modules/IO/DCMTK/test/itkDCMTKGetDicomTagsTest.cxx

```diff
@@ -0,0 +1,215 @@
+  {
+    // YYYYMMDD
+    outputFile << strTarget << std::endl;
+    std::string expectedVal = "20030625";
+    ITK_TRY_EXPECT_EQUAL(strTarget, expectedVal);
+  }
+
+  // Study Time
+  group = 0x0008;
+  element = 0x0030;
+  // seq.GetElementTM(group, element, tag)
+  if (fileReader.GetElementTM(group, element, strTarget, throwException) == EXIT_SUCCESS)
+  {
+    outputFile << strTarget << std::endl;
+    std::string expectedVal = "152734";
+    ITK_TRY_EXPECT_EQUAL(strTarget, expectedVal);
+  }
+
+  // Patient Name
+  group = 0x0010;
+  element = 0x0010;
+  if (fileReader.GetElementPN(group, element, strTarget, throwException) == EXIT_SUCCESS)
+  {
+    outputFile << strTarget << std::endl;
+    std::string expectedVal = "Wes Turner";
+    ITK_TRY_EXPECT_EQUAL(strTarget, expectedVal);
+  }
+
+  // Patient Sex
+  group = 0x0010;
+  element = 0x0040;
+  if (fileReader.GetElementCS(group, element, strTarget, throwException) == EXIT_SUCCESS)
+  {
+    outputFile << strTarget << std::endl;
+    std::string expectedVal = "O";
+    ITK_TRY_EXPECT_EQUAL(strTarget, expectedVal);
+  }
+
+  // Patient Weight
+  group = 0x0010;
+  element = 0x1030;
+  if (fileReader.GetElementDS(group, element, 1, strTarget, throwException) == EXIT_SUCCESS)
+  {
+    outputFile << strTarget << std::endl;
+    std::string expectedVal = "68.039";
+    ITK_TRY_EXPECT_EQUAL(strTarget, expectedVal);
+  }
+
+  // Echo Number
+  group = 0x0018;
+  element = 0x0086;
+  itk::int32_t int32Target;
+  if (fileReader.GetElementIS(group, element, int32Target, throwException) == EXIT_SUCCESS)
+  {
+    outputFile << int32Target << std::endl;
+    itk::int32_t expectedVal = 12576; // Expect (12576)"1"
+    ITK_TRY_EXPECT_EQUAL(int32Target, expectedVal);
+  }
+
+  // Acquisition Duration
+  group = 0x0019;
+  element = 0x105a;
+  float floatTarget;
+  if (fileReader.GetElementFL(group, element, floatTarget, throwException) == EXIT_SUCCESS)
+  {
+    outputFile << floatTarget << std::endl;
+    float expectedVal = 1304791694;
+    ITK_TRY_EXPECT_EQUAL(floatTarget, expectedVal);
+  }
+
+  // Locs per 3D slab
+  group = 0x0021;
+  element = 0x1057;
+  if (fileReader.GetElementSL(group, element, int32Target, throwException) == EXIT_SUCCESS)
+  {
+    outputFile << int32Target << std::endl;
+    itk::int32_t expectedVal = 124; // Expect (124)"|"
+    ITK_TRY_EXPECT_EQUAL(int32Target, expectedVal);
+  }
+
+  // Samples per Pixel
+  group = 0x0028;
+  element = 0x0002;
+  if (fileReader.GetElementUS(group, element, usTarget, throwException) == EXIT_SUCCESS)
+  {
+    outputFile << usTarget << std::endl;
+    unsigned short expectedVal = 1; // for scalar; 3 for RGB
+    ITK_TRY_EXPECT_EQUAL(usTarget, expectedVal);
+  }
+
+  // Photometric Interpretation
+  group = 0x0028;
+  element = 0x0004;
+  if (fileReader.GetElementUS(group, element, strTarget, throwException) == EXIT_SUCCESS)
+  {
+    outputFile << strTarget << std::endl;
+    unsigned short expectedVal = "????";
```

Comment:
PhotometricInterpretation is VR CS (Code String), not US.

Reply:
Sorry, my bad. Fixed in 172680e. Thanks for the review.

Reply:
Thank you. The line
`unsigned short expectedVal = "????";`
will not compile.

Reply:
I know. Thanks for noticing. Following https://github.com/InsightSoftwareConsortium/ITK/pull/2796#discussion_r1014594203 all values should be checked, but it will take before I am able to do that. Ideally I'd also remove the expected values and the comparisons from the test, and rely on CMake and the text files for quantitative checking purposes.